interactive/model_driver/session_model_runner.py

```diff
@@ -0,0 +1,379 @@
+# SESSION BASED MODEL RUNNING
+
+
+from django.shortcuts import render
+from django.http import JsonResponse, HttpResponse, HttpResponseBadRequest
+from django.conf import settings
+import json
+import os
+import subprocess
+import mimetypes
+import pandas as pd
+import time
+from shutil import copy
+from .run_setup import setup_run
+# from .run_logging import *
+from datetime import datetime
+from mechanism.reactions import reactions_are_valid
+from interactive.tools import *
+from pathlib import Path
+
+class SessionModelRunner():
+    def __init__(self, session_id):
+        self.setPathsForSessionID(session_id)
+    interface_solo = False
+    if "MUSIC_BOX_BUILD_DIR" in os.environ:
+        mb_dir = os.path.join(os.environ['MUSIC_BOX_BUILD_DIR'])
+        interface_solo = False
+    else:
+        print(os.environ)
+        mb_dir = ''
+        interface_solo = True
+
+    out_path = os.path.join(mb_dir, 'output.csv')
+    error_path = os.path.join(mb_dir, 'error.json')
+    copy_path = os.path.join(settings.BASE_DIR, 'dashboard/static/past_run/past.csv')
+    config_path = os.path.join(settings.BASE_DIR, "dashboard/static/config/my_config.json")
+    old_path = os.path.join(settings.BASE_DIR, "dashboard/static/config/old_config.json")
+    complete_path = os.path.join(mb_dir, 'MODEL_RUN_COMPLETE')
+
+
+    config_dest = os.path.join(settings.BASE_DIR, 'dashboard/static/past_run/config.json')
+
+    config_folder_path = os.path.join(settings.BASE_DIR, "dashboard/static/config")
+    log_path = os.path.join(settings.BASE_DIR, 'dashboard/static/log/log')
+    camp_folder_path = os.path.join(settings.BASE_DIR, "dashboard/static/config/camp_data")
+
+    reactions_path = os.path.join(settings.BASE_DIR, "dashboard/static/config/camp_data/reactions.json")
+    species_path = os.path.join(settings.BASE_DIR, "dashboard/static/config/camp_data/species.json")
+
+    sessionid = ""
+    def setPathsForSessionID(self, session_id):
+        global mb_dir
+        global out_path
+        global error_path
+        global copy_path
+        global config_path
+        global old_path
+        global complete_path
+        global config_dest
+        global config_folder_path
+        global camp_folder_path
+        global reactions_path
+        global species_path
+        global sessionid
+        global log_path
```

Comment:
How is using so many global variables on a shared server considered multiuser?

Reply:
they're global variables within a class so it won't affect other users

Reply:
a new class is made everytime a user runs a model so these global variables cant affect other classes made by other users

Reply:
Ah, then nevermind